code/game/machinery/teleporter.dm

```diff
@@ -370,6 +370,70 @@
 	else
 		icon_state = "tele0"
 
+/obj/machinery/teleport/perma
+	name = "permanent teleporter"
+	desc = "A teleporter with the target pre-set on the circuit board."
+	icon_state = "tele0"
+	var/recalibrating = 0
+	use_power = 1
+	idle_power_usage = 10
+	active_power_usage = 2000
+	
+	var/target
+	var/tele_delay = 50
+
+/obj/machinery/teleport/perma/RefreshParts()
+	for(var/obj/item/weapon/circuitboard/teleporter_perma/C in component_parts)
+		target = C.target
+	var/A = 40
+	for(var/obj/item/weapon/stock_parts/matter_bin/M in component_parts)
+		A -= M.rating * 10
+	tele_delay = max(A, 0)
+	update_icon()
+
+/obj/machinery/teleport/perma/Bumped(M as mob|obj)
```

Comment:
I'd advocate that you make this a child of the hub, but then I note that this may cause issues with teleconsoles trying to link to it